CesiumRasterOverlays/src/RasterizedPolygonsOverlay.cpp

```diff
@@ -83,88 +86,38 @@ void rasterizePolygons(
-    for (size_t triangle = 0; triangle < indices.size() / 3; ++triangle) {
-      const glm::dvec2& a = vertices[indices[3 * triangle]];
-      const glm::dvec2& b = vertices[indices[3 * triangle + 1]];
-      const glm::dvec2& c = vertices[indices[3 * triangle + 2]];
-
-      // TODO: deal with the corner cases here
-      const double minX = glm::min(a.x, glm::min(b.x, c.x));
-      const double minY = glm::min(a.y, glm::min(b.y, c.y));
-      const double maxX = glm::max(a.x, glm::max(b.x, c.x));
-      const double maxY = glm::max(a.y, glm::max(b.y, c.y));
-
-      const CesiumGeospatial::GlobeRectangle triangleBounds(
-          minX,
-          minY,
-          maxX,
-          maxY);
-
-      // skip this triangle if it is entirely outside the tile bounds
-      if (!rectangle.computeIntersection(triangleBounds)) {
-        continue;
-      }
-
-      const glm::dvec2 ab = b - a;
-      const glm::dvec2 ab_perp(-ab.y, ab.x);
-      const glm::dvec2 bc = c - b;
-      const glm::dvec2 bc_perp(-bc.y, bc.x);
-      const glm::dvec2 ca = a - c;
-      const glm::dvec2 ca_perp(-ca.y, ca.x);
-
-      size_t width = size_t(image.width);
-      size_t height = size_t(image.height);
-
-      for (size_t j = 0; j < height; ++j) {
-        const double pixelY =
-            rectangle.getSouth() +
-            rectangleHeight * (1.0 - (double(j) + 0.5) / double(height));
-        for (size_t i = 0; i < width; ++i) {
-          const double pixelX = rectangle.getWest() + rectangleWidth *
-                                                          (double(i) + 0.5) /
-                                                          double(width);
-          const glm::dvec2 v(pixelX, pixelY);
-
-          const glm::dvec2 av = v - a;
-          const glm::dvec2 cv = v - c;
-
-          const double v_proj_ab_perp = glm::dot(av, ab_perp);
-          const double v_proj_bc_perp = glm::dot(cv, bc_perp);
-          const double v_proj_ca_perp = glm::dot(cv, ca_perp);
-
-          // will determine in or out, irrespective of winding
-          if ((v_proj_ab_perp >= 0.0 && v_proj_ca_perp >= 0.0 &&
-               v_proj_bc_perp >= 0.0) ||
-              (v_proj_ab_perp <= 0.0 && v_proj_ca_perp <= 0.0 &&
-               v_proj_bc_perp <= 0.0)) {
-            image.pixelData[width * j + i] = insideColor;
-          }
-        }
-      }
-    }
+  image.pixelData.resize(
+      size_t(image.width * image.height * 4),
+      std::byte{outsideColor});
+
+  CesiumVectorData::VectorRasterizer rasterizer(
+      rectangle,
+      loaded.pImage,
+      0,
+      ellipsoid);
+  rasterizer.clear(CesiumUtility::Color{outsideColor, 0x00, 0x00, 0xff});
+
+  CesiumVectorData::PolygonStyle insideStyle{
+      CesiumVectorData::ColorStyle{
+          CesiumUtility::Color{insideColor, 0x00, 0x00, 0xff}},
+      std::nullopt};
+  for (const CartographicPolygon& selection : cartographicPolygons) {
+    rasterizer.drawPolygon(selection, insideStyle);
```

Comment:
This new implementation is probably much faster than the old one too, right? Worth mentioning in the changelog, if so.